### PR TITLE
Animals obey conservation of matter unless they are undead

### PR DESCRIPTION
**Pull request by Sirionaut**
## About the PR
Somewhat addresses #21726 & #18927

- Animals only produce if their hunger level is at least **okay** (before it was **peckish**)
- They now also consume extra nutrients when doing that (before it was only chickens)
- Hungerless animals (zombies) still produce endlessly, eggLayer code implied that was on purpose and I love it, so I expanded it
- Zombification removes reproductive abilities
- Refactored a bit
 
does NOT include:
- shared animal metabolism update thing

## Why / Balance
Animals should obey the 1st law of thermodynamics, unless they are zombies.

I like the idea of zombie animals still producing food, it allows for FUN when someone lets loose the zombified food supply.

eggLayer: One every 60-120 seconds for 60 hunger (unchanged)
udder: 25 every 60 seconds for 10 hunger
wool: 25 every 60 seconds for 10 hunger

While testing, I noticed that zombie animals can procreate with healthy ones.
Zombies should reproduce via biting, not traditional biology.

## Technical details

- `ZombieSystem.Transform.cs` removes reproductive components
- Animalsystems now check `_mobState.IsDead(uid)` 
- Consumes nutrients when producing (if the entity has hunger)
- changed udders to use `base.update(frameTime)` instead of counting frames like before. (I guessed this is the newer and better way to do it?)

## Media

- [X] this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:

- fix: Animals now require a more reasonable amount of satiation before producing eggs, milk, or wool (unless they are undead)

### CODE REVIEW

**Comment by Sirionaut** [comment]
last updates is me trying to implement what was written in 
https://github.com/space-wizards/space-station-14/discussions/21113 &
https://github.com/space-wizards/space-station-14/discussions/20585

File: Content.Server/Animals/Components/EggLayerComponent.cs [hunk 1]
```diff
@@ -1,36 +1,35 @@
...
 ///     This component handles animals which lay eggs (or some other item) on a timer, using up hunger to do so.
 ///     It also grants an action to players who are controlling these entities, allowing them to do it manually.
 /// </summary>
+
 [RegisterComponent]
 public sealed partial class EggLayerComponent : Component
 {
-    [DataField("eggLayAction", customTypeSerializer: typeof(PrototypeIdSerializer<EntityPrototype>))]
+    [DataField]
     public string EggLayAction = "ActionAnimalLayEgg";
```
Comment (deltanedas) [inline Content.Server/Animals/Components/EggLayerComponent.cs:15]: EntProtoId

**Review by deltanedas — CHANGES_REQUESTED** [review]

File: Content.Server/Animals/Components/EggLayerComponent.cs (hunk 1 above)
Comment (deltanedas) [reply Content.Server/Animals/Components/EggLayerComponent.cs]: if you remove the serializer it wont catch invalid ids
EntProtoId does this automatically while string needs the type serializer

**Review by mirrorcult — APPROVED** [review]
lovely PR!!

**Comment by mirrorcult** [comment]
going to modify the changelog here, since i think its not clear enough what the PR changes

**Pull request merged**